include/alpaka/kernel/Traits.hpp

```diff
@@ -202,6 +202,29 @@ namespace alpaka
         "-Wdocumentation" // clang does not support the syntax for variadic template arguments "args,..."
 #endif
 
+
+    /// Check if a type used as kernel argument is trivially copyable
+    ///
+    /// \attention In case this trait is specialized for a user type the user should be sure that the result of calling
+    /// the copy constructor is equal to use memcopy to duplicate the object. An existing destructor should be side
+    /// effect free.
+    ///
+    /// It's implementation defined whether the closure type of a lambda is trivially copyable.
+    /// Therefor the default implementation is true for trivially copyable or empty (stateless) types.
+    ///
+    /// @tparam T type to check
+    /// @{
+    template<typename T>
+    struct IsKernelArgumentTriviallyCopyable
+        : std::bool_constant<std::is_empty_v<T> || std::is_trivially_copyable_v<T>>
+    {
+    };
+
+    template<typename T>
+    constexpr bool IsKernelArgumentTriviallyCopyable_v = IsKernelArgumentTriviallyCopyable<T>::value;
```

Comment:
Following the styles of the other alpaka traits (like `isQueue<T>`, _etc_.) this should probably be
```suggestion
    inline constexpr bool isKernelArgumentTriviallyCopyable = IsKernelArgumentTriviallyCopyable<T>::value;
```

Reply:
Agreed.

Reply:
done